Disable statx for all builds.

libstd implements statx using syscall and thus it will not work on
scratchbox2. Moreover as it is supported only by Linux 4.11 and newer
it's little use for us at the moment. It can be disabled and which means
that optional extra information is set to None. Software should handle
that gracefully without issues thanks to rust's type system.

---
--- rustc-1.44.0-src/src/libstd/sys/unix/fs.rs.old	2020-06-01 18:44:16.000000000 +0300
+++ rustc-1.44.0-src/src/libstd/sys/unix/fs.rs	2020-06-26 12:42:46.438070586 +0300
@@ -58,20 +58,9 @@
 // https://github.com/rust-lang/rust/pull/67774
 macro_rules! cfg_has_statx {
     ({ $($then_tt:tt)* } else { $($else_tt:tt)* }) => {
-        cfg_if::cfg_if! {
-            if #[cfg(all(target_os = "linux", target_env = "gnu"))] {
-                $($then_tt)*
-            } else {
-                $($else_tt)*
-            }
-        }
-    };
-    ($($block_inner:tt)*) => {
-        #[cfg(all(target_os = "linux", target_env = "gnu"))]
-        {
-            $($block_inner)*
-        }
+        $($else_tt)*
     };
+    ($($block_inner:tt)*) => {};
 }
 
 cfg_has_statx! {{
